src/compiler/parse/state/tag.ts

```diff
@@ -189,9 +189,7 @@ export default function tag(parser: Parser) {
 		}
 
 		const definition = element.attributes.splice(index, 1)[0];
-		if (definition.value === true
-			|| definition.value.length !== 1
-			|| !['Text', 'MustacheTag', 'AttributeShorthand'].includes(definition.value[0].type)) {
+		if (definition.value === true	|| !['Text', 'MustacheTag', 'AttributeShorthand'].includes(definition.value[0].type)) {
```

Comment:
Shouldn't this now validate all elements of the array not just the 0th?

Reply:
actually, `['Text', 'MustacheTag', 'AttributeShorthand']` seems like its all the possible values for `Attribute#value`, so maybe we dont need this extra check?

at the same time, i wonder maybe we just validate against `AttributeShorthand`? coz i dont think it make sense to have

```svelte
<svelte:element {this} />
```

since you cant possibly declare a variable named `this`

Reply:
> actually, ['Text', 'MustacheTag', 'AttributeShorthand'] seems like its all the possible values for Attribute#value, so maybe we dont need this extra check?

I rethink it and I agree that we don't need to check it.

> at the same time, i wonder maybe we just validate against AttributeShorthand? coz i dont think it make sense to have

Yes. You are right.

I removed the validation in this commit.
https://github.com/sveltejs/svelte/pull/6898/commits/c5c0dee90e12519cd3b07f753e8fc7c744032a20

Reply:
thanks :)

probably also add a similar validation test same as above?

test if we throw error when the value is `true`

```svelte
<svelte:element this />
```

Reply:
The error message is not related to `<svelte:element>`.
So I didn't add the test but I added the test because this test is meaningful.
https://github.com/sveltejs/svelte/pull/6898/commits/2858cf582ee25ab216148c3cfb74d34ed20ab6e3

![image](https://user-images.githubusercontent.com/19153718/161900990-c7c72354-2441-48d3-8cc8-0d167e21213e.png)